test/e2e/framework/pod/resize.go

```diff
@@ -189,54 +190,119 @@ func MakePodWithResizableContainers(ns, name, timeStamp string, tcInfo []Resizab
 			},
 		},
 		Spec: v1.PodSpec{
-			OS:            &v1.PodOS{Name: v1.Linux},
-			Containers:    testContainers,
-			RestartPolicy: v1.RestartPolicyOnFailure,
+			OS:             &v1.PodOS{Name: v1.Linux},
+			InitContainers: testInitContainers,
+			Containers:     testContainers,
+			RestartPolicy:  v1.RestartPolicyOnFailure,
 		},
 	}
 	return pod
 }
 
-func VerifyPodResizePolicy(gotPod *v1.Pod, wantCtrs []ResizableContainerInfo) {
+// separateContainers splits the input into initContainers and normal containers.
+func separateContainers(tcInfo []ResizableContainerInfo) ([]v1.Container, []v1.Container) {
+	var initContainers, containers []v1.Container
+
+	for _, ci := range tcInfo {
+		tc := makeResizableContainer(ci)
+		if ci.InitCtr {
+			initContainers = append(initContainers, tc)
+		} else {
+			containers = append(containers, tc)
+		}
+	}
+
+	return initContainers, containers
+}
+
+// separateContainerStatuses splits the input into initContainerStatuses and containerStatuses.
+func separateContainerStatuses(tcInfo []ResizableContainerInfo) ([]v1.ContainerStatus, []v1.ContainerStatus) {
+	var containerStatuses, initContainerStatuses []v1.ContainerStatus
+
+	for _, ci := range tcInfo {
+		ctrStatus := v1.ContainerStatus{
+			Name:         ci.Name,
+			RestartCount: ci.RestartCount,
+		}
+		if ci.InitCtr {
+			initContainerStatuses = append(initContainerStatuses, ctrStatus)
+		} else {
+			containerStatuses = append(containerStatuses, ctrStatus)
+		}
+	}
+
+	return initContainerStatuses, containerStatuses
+}
+
+func VerifyPodResizePolicy(gotPod *v1.Pod, wantInfo []ResizableContainerInfo) {
+	ginkgo.GinkgoHelper()
+
+	wantInitCtrs, wantCtrs := separateContainers(wantInfo)
+	verifyPodContainersResizePolicy(gotPod.Spec.InitContainers, wantInitCtrs)
+	verifyPodContainersResizePolicy(gotPod.Spec.Containers, wantCtrs)
+}
+
+func verifyPodContainersResizePolicy(gotCtrs []v1.Container, wantCtrs []v1.Container) {
 	ginkgo.GinkgoHelper()
-	gomega.Expect(gotPod.Spec.Containers).To(gomega.HaveLen(len(wantCtrs)), "number of containers in pod spec should match")
+	gomega.Expect(gotCtrs).To(gomega.HaveLen(len(wantCtrs)), "number of containers in pod spec should match")
+
 	for i, wantCtr := range wantCtrs {
-		gotCtr := &gotPod.Spec.Containers[i]
-		ctr := makeResizableContainer(wantCtr)
-		gomega.Expect(gotCtr.Name).To(gomega.Equal(ctr.Name))
-		gomega.Expect(gotCtr.ResizePolicy).To(gomega.Equal(ctr.ResizePolicy))
+		gotCtr := gotCtrs[i]
```

Comment:
It seems a little weird that we're assuming the containers are in the same order after splitting the list of containers. Maybe it would be better to just match the container by name here, and then you don't actually need to split the `wantInfos`. You can combine the containers & init containers into a single list to iterate over, or just use `podutil.VisitContainers`.